Automattic/wp-calypso
# Signup: Add signup flow for upgrading a domain-only site

# This is a replacement for #11224 which was broken for the main flow.
It is different only by the last commit.

This PR aims to allow users to upgrade their "domain only site" to a regular site with plan.
This PR also fixes a few bugs related to previous "featuregate" work ( the first 2 commits ).

#### Testing instructions
  
Run `git checkout add/upgrade-domain-only` and start your server, or open a [live branch](https://calypso.live/?branch=add/upgrade-domain-only)

Flow:
0. Click "Create a new site" on a "domain only site" ( you might want to create one of those first ).
![screen shot 2017-02-07 at 14 30 17](https://cloud.githubusercontent.com/assets/326402/22691212/05849230-ed42-11e6-9429-95141b119827.png)


1. Choose a theme
![screen shot 2017-02-07 at 14 28 01](https://cloud.githubusercontent.com/assets/326402/22691135/af290218-ed41-11e6-8f63-098ddd53b92d.png)

2. Choose a plan ( no free plan )
![screen shot 2017-02-07 at 14 25 26](https://cloud.githubusercontent.com/assets/326402/22691103/7b91cade-ed41-11e6-90a3-9d1ffda06f7b.png)

3. Checkout ( with discounted price due to existing domain )
![screen shot 2017-02-07 at 14 01 11](https://cloud.githubusercontent.com/assets/326402/22691105/7f9db4d0-ed41-11e6-9b82-e7e61f6c35fa.png)

4. Thank you screen AND Site has a plan + no `is_domain_only` option, that requires the server side patch ( D4297 ).
![screen shot 2017-02-07 at 16 48 03](https://cloud.githubusercontent.com/assets/326402/22696027/6268dda4-ed55-11e6-98f8-683b296c12d2.png)

5. Start over again and make sure sign up flow work for default use case and free .wordpress.com subdomain.

#### Reviews
